src/pages/quiz/creation/SingleChoiceQuestionView.tsx

```diff
@@ -0,0 +1,127 @@
+import React from "react";
+import {StyleSheet, Text, View} from "react-native";
+import BouncyCheckbox from "react-native-bouncy-checkbox";
+import {SingleChoiceQuestion} from "./Question";
+
+interface SingleChoiceQuestionViewProps {
+    question: SingleChoiceQuestion;
+    userAnswer: number;
+    setUserAnswer: (userAnswer: number) => void;
+    isEditable: boolean;
+}
+
+const SingleChoiceQuestionView: React.FC<SingleChoiceQuestionViewProps> = ({
+                                                                               question,
+                                                                               userAnswer,
+                                                                               setUserAnswer,
+                                                                               isEditable,
+                                                                           }) => {
+    const handleOptionToggle = (index: number) => {
+        setUserAnswer(index);
+    };
+
+    return (
+        <View style={styles.container}>
+            {question.choices.map((choice, index) => (
+                <View
+                    style={[
+                        styles.choiceContainer
+                    ]}
+                    key={index}
+                >
+                    <View
+                        style={[
+                            styles.checkBoxContainer
+                        ]}
+                        key={index}
+                    >
+                        <img
+                            style={{
+                                marginRight: !isEditable && userAnswer === index && question.answer === userAnswer ? 20 : 0,
```

Comment:
I know this is improvement if this works but I would define just two styles for this icon and then write only one condition. 
```style={(!isEditable && userAnswer === index && question.answer === userAnswer)? styles.highlightedStyle : styles.defaultStyle}```

Reply:
great idea